src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const transformedState = Object.assign({}, state);
+  const transformedArray = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(transformedState, action.extraData);
```

Comment:
You can make destructuring to get all the properties of action that you need.
It will make your code cleaner